### PR TITLE
Add Rails 2.7.1, Node 10.21.0, Yarn 1.22.4

### PR DESCRIPTION
**Pull request by sixfootsixdesigns**
adding rails 2.7.1 for sponsorship since we use the newer rails stuff

### CODE REVIEW

File: python-serverless/2/Dockerfile [hunk 1]
```diff
@@ -30,3 +30,4 @@ RUN curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-${CC_T
     pip install -U --user aws-cdk.aws_sns && \
     pip install -U --user aws-cdk.aws_sqs && \
     pip install -U --user aws-cdk.aws_ssm
```
Comment (wingrunr21) [inline python-serverless/2/Dockerfile:32]: Missed a `\` at the end of the line

Comment (sixfootsixdesigns) [reply python-serverless/2/Dockerfile]: fixed

File: python-serverless/2/Dockerfile [hunk 2]
```diff
@@ -29,4 +29,5 @@ RUN curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-${CC_T
     pip install -U --user aws-cdk.aws_secretsmanager && \
     pip install -U --user aws-cdk.aws_sns && \
     pip install -U --user aws-cdk.aws_sqs && \
-    pip install -U --user aws-cdk.aws_ssm
+    pip install -U --user aws-cdk.aws_ssm && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (wingrunr21) [inline python-serverless/2/Dockerfile:33]: Looks like based on the build you need a `sudo` here as well:

`sudo rm -rf /var/lib/apt/lists/*`

Comment (sixfootsixdesigns) [reply python-serverless/2/Dockerfile]: fixed as well

File: python-serverless/2/Dockerfile [hunk 3]
```diff
@@ -5,7 +5,7 @@ ENV CC_TEST_REPORTER_VERSION=0.6.2
 RUN curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-${CC_TEST_REPORTER_VERSION}-linux-amd64 > ~/cc-test-reporter && \
     chmod +x ~/cc-test-reporter && \
     sudo mv ~/cc-test-reporter /usr/local/bin/cc-test-reporter && \
-    sudo apt-get install nodejs && \
+    sudo apt-get update && sudo apt-get install nodejs && \
```
Comment (edbarnesG) [inline python-serverless/2/Dockerfile:8]: Probably could make that separate commands for clarity. Also with this change I can't 100% for sure say it won't break existing projects without testing all of them, so if it's really necessary it's worth considering making a python-serverless/3 container with this change and current versions of awscli and aws-sam-cli and maybe boto3. If it's not really necessary then I'd call it a "not broke, don't fix it" situation.

Comment (wingrunr21) [reply python-serverless/2/Dockerfile:8]: `aws-cdk`'s node versions say 10.3 and up. The fact this is broken is blocking this PR

Comment (sixfootsixdesigns) [reply python-serverless/2/Dockerfile:8]: it's broken because the version of node it was asking for doesn't exist anymore.

Comment (sixfootsixdesigns) [reply python-serverless/2/Dockerfile:8]: node 10.21.0 is now used instead of node 10.15.2

**Pull request merged**